diff --git a/libglycin/meson.build b/libglycin/meson.build
index 3192e06..6c164a3 100644
--- a/libglycin/meson.build
+++ b/libglycin/meson.build
@@ -18,7 +18,7 @@ cargo_options = [
     '--manifest-path', manifest,
 ]
 
-pkg_uninstalled_path = {'PKG_CONFIG_PATH': meson.project_build_root() / 'meson-uninstalled'}
+pkg_uninstalled_path = {}
 
 cargo_env = {
     'CARGO_HOME': cargo_home,
@@ -35,7 +35,7 @@ packages = [
         'suffix': '-gtk4',
         'namespace_suffix': 'Gtk4',
         'symbol_prefix_suffix': '_gtk',
-        'extra_deps': [dependency('gtk4', version: gtk4_req)],
+        'extra_deps': [dependency('gtk4', version: gtk4_req), dependency('glycin-1')],
     },
 ]
 
@@ -167,7 +167,7 @@ foreach package : packages
     if get_option('vapi')
         # Generate .vapi (Vala)
         gnome.generate_vapi(
-            f'libglycin@suffix@-1',
+            f'glycin@suffix@-1',
             sources: libglycin_gir[0],
             packages: ['glib-2.0', 'gobject-2.0', 'gio-2.0', 'gtk4'],
             gir_dirs: meson.current_build_dir(),
